File: app/src/main/java/com/bopr/android/smailer/ui/PhoneFilterListFragment.java
package com.bopr.android.smailer.ui;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.bopr.android.smailer.R;

/**
 * Phone filter list activity fragment.
 *
 * @author Boris Pronin (<a href="mailto:boprsoft.dev@gmail.com">boprsoft.dev@gmail.com</a>)
 */
abstract class PhoneFilterListFragment extends FilterListFragment {

    @NonNull
    @Override
    EditFilterListItemDialogFragment createEditItemDialog(@Nullable String text) {
        EditPhoneDialogFragment dialog = new EditPhoneDialogFragment();
        dialog.setTitle(text == null ? R.string.add : R.string.edit);
        dialog.setInitialValue(text);
        return dialog;
    }

    @Nullable
    @Override
    String getItemText(@Nullable String value) {
        return value;
    }
}

File: app/src/main/kotlin/com/bopr/android/smailer/ui/PhoneFilterListFragment.kt
package com.bopr.android.smailer.ui

import com.bopr.android.smailer.R

/**
 * Phone filter list activity fragment.
 *
 * @author Boris Pronin ([boprsoft.dev@gmail.com](mailto:boprsoft.dev@gmail.com))
 */
abstract class PhoneFilterListFragment : FilterListFragment() {

    override fun createEditItemDialog(text: String?): BaseEditDialogFragment<String> {
        val dialog = EditPhoneDialogFragment()
        dialog.setTitle(if (text == null) R.string.add else R.string.edit)
        dialog.setValue(text)
        return dialog
    }

    override fun getItemText(value: String?): String? {
        return value
    }
}